File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        parenthesis={')':'(', '}':'{',  ']':'['  }
        stack=[]
        for par in s:
            if par in parenthesis:
                if stack and stack[-1]==parenthesis[par]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(par)
        return True if not stack else False
            
        

File: 20-valid-parentheses/valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let closing=new Map()
    closing.set("}","{")
    closing.set("]","[")
    closing.set(")","(")
    let res=[]
    for(bracket of s){
        if(closing.has(bracket)&&res[res.length-1]===closing.get(bracket)){
            res.pop();
        }else{
            res.push(bracket);
        }
    }
    console.log(res.length)
    return res.length===0;

};